array_analyzer/extract/img_processing.py

```diff
@@ -252,28 +254,47 @@ def crop_image(arr, cx_, cy_, radius_, border_=200):
     return crop
 
 
-def thresh_and_binarize(image_, method='rosin', invert=True, min_size=10, thr_percent=95):
+def get_largest_component(spot_segm):
+    """
+    Remove everything but the largest connected component from segmented image.
+
+    :param np.array spot_segm: Binary segmented 2D image
+    :return np.array largest_component: Largest connected component in image
+    """
+    labels = label(spot_segm)
+    largest_component = labels.copy()
+    if labels.max() > 0:
```

Comment:
This is minor but I think you can do `if not spot_segm.any(): return spot_segm` to skip labeling the image if the image is all zeros